vnext/Microsoft.ReactNative/Utils/ImageUtils.cpp

```diff
@@ -23,13 +23,62 @@ winrt::IAsyncOperation<winrt::IRandomAccessStream> GetImageStreamAsync(ReactImag
   try {
     co_await winrt::resume_background();
 
-    auto httpMethod{
-        source.method.empty() ? winrt::HttpMethod::Get() : winrt::HttpMethod{winrt::to_hstring(source.method)}};
-
     winrt::Uri uri = UriTryCreate(winrt::to_hstring(source.uri));
     if (!uri) {
       co_return nullptr;
     }
+
+    if (uri.SchemeName() == L"file") {
+      auto path = winrt::to_string(uri.Path());
+      std::replace(path.begin(), path.end(), '/', '\\');
+      auto getFileSync = winrt::Windows::Storage::StorageFile::GetFileFromPathAsync(winrt::to_hstring(path));
+      co_await lessthrow_await_adapter<
+          winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>{getFileSync};
+
+      if (FAILED(getFileSync.ErrorCode())) {
+        co_return nullptr;
+      }
+
+      winrt::Windows::Storage::StorageFile file{getFileSync.GetResults()};
+
+      auto openReadAsync = file.OpenReadAsync();
+      co_await lessthrow_await_adapter<winrt::Windows::Foundation::IAsyncOperation<
+          winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>{openReadAsync};
+
+      if (FAILED(openReadAsync.ErrorCode())) {
+        co_return nullptr;
+      }
+
+      winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType stream{openReadAsync.GetResults()};
+      return stream;
+    }
+
+    if (uri.SchemeName() == L"ms-appx") {
+      auto getFileSync = winrt::Windows::Storage::StorageFile::GetFileFromApplicationUriAsync(uri);
+      co_await lessthrow_await_adapter<
+          winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Storage::StorageFile>>{getFileSync};
+
+      if (FAILED(getFileSync.ErrorCode())) {
+        co_return nullptr;
+      }
+
+      winrt::Windows::Storage::StorageFile file{getFileSync.GetResults()};
+
+      auto openReadAsync = file.OpenReadAsync();
+      co_await lessthrow_await_adapter<winrt::Windows::Foundation::IAsyncOperation<
+          winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType>>{openReadAsync};
+
+      if (FAILED(openReadAsync.ErrorCode())) {
+        co_return nullptr;
+      }
+
+      winrt::Windows::Storage::Streams::IRandomAccessStreamWithContentType stream{openReadAsync.GetResults()};
+      return stream;
```

Comment:
Thise whole section is identical with the above. Instead you could have something more like:

```
bool isFile = (uri.SchemeName() == L"file");
bool isAppx = (uri.SchemeName() == L"ms-appx");

if (isFile || isAppx) {
  IAsyncOperation<StorageFile> getFileSync;

  if (isFile) {
    // blah
  } else {
    getFileSync = winrt::Windows::Storage::StorageFile::GetFileFromApplicationUriAsync(uri);
  }
  
  // rest goes here
}
```